tasks/7.txt

```diff
@@ -4,4 +4,27 @@ Create a query to return all employees and full name of person to whom this empl
 
 Full Name - title of courtesy with full name.
 Order data by first column.
-Reports To - Full name. If the employee does not report to anybody leave "-" in the column.
\ No newline at end of file
+Reports To - Full name. If the employee does not report to anybody leave "-" in the column.
+
+db.employees.aggregate([
+    {$lookup: {
+            from: "employees",
+            localField: "ReportsTo",
+            foreignField: "EmployeeID",
+            as: "inventory_docs"
+        }
+    },
+    {$project:{
+        _id:0,"EmployeeID":1,
+        "FullName":{$concat:["$TitleOfCourtesy"," ","$FirstName"," ","$LastName"]},
+         "Reports To":
+         {$ifNull:[
+         {$concat: [
+         {$arrayElemAt: ["$inventory_docs.TitleOfCourtesy", 0]},
+         " ",
+         {$arrayElemAt: ["$inventory_docs.FirstName", 0]},
+         " ",
+         {$arrayElemAt: ["$inventory_docs.LastName", 0]}]},
+         "-"]}
+        }}
+])
```

Comment:
For 2 records FullName will be a null in the solution, but it is not expected behaviour

Reply:
So that, I have changed collection fields whith "FirstName"==="NULL" to a really null.
And then I am checking if  " FirstName"===null  it will be replaced with "" in both cases ("FullName" and "Reports To").